src/Microsoft.ML.Data/Data/Conversion.cs

```diff
@@ -1585,6 +1585,17 @@ public void Convert(in TX span, ref I4 value)
         {
             value = ParseI4(in span);
         }
+        public bool TryConvert(in TX span, ref I4 value)
```

Comment:
Why not use the existing `TryParse` method?

https://github.com/dotnet/machinelearning/blob/369a9b614b68793de8401c843df6f5d566088993/src/Microsoft.ML.Data/Data/Conversion.cs#L1184-L1201

Reply:
I was trying to remain consistent with the existing code, which assigned a value, then passed it by ref, and then checked whether it remained what it was assigned.  In practice it wouldn't matter because it would end up with the default value in either case, but it would incur additional writes in the failure case, and I didn't know whether it would matter.  When I don't know a code base particularly well and what ramifications my changes might have, I try to maintain the existing code as much as possible.

Reply:
I guess my concern is the consistency in the internal `Conversion` class. 

1. We now have `TryParse` and `TryConvert`.  Which one should new callers choose?
2. We have `TryConvert` for Int32, but no other types?

Reply:
1. Whichever signature they need. If they want an out, they choose TryParse, if they want ref, they choose TryConvert.
2. Yes.  You'd prefer we add a bunch of dead code?

If you feel strongly about this and want to change the call site to use TryParse, go for it.  As I noted, my goal was to muck minimally with the call site logic.